MetadataProcessor.Shared/Utility/nanoMetadataToken.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Mono.Cecil;
+
+namespace nanoFramework.Tools.MetadataProcessor
+{
+    /// <summary>
+    /// Encoded type for inline type calls.
+    /// </summary>
+    public class nanoMetadataToken
+    {
+        private NanoClrTable _clrTable;
+        private ushort _id;
+
+        public nanoMetadataToken()
+        {
+        }
+
+        public nanoMetadataToken(NanoClrTable clrTable, ushort id)
+        {
+            _clrTable = clrTable;
+            _id = id;
+        }
+
+        public nanoMetadataToken(MetadataToken token, ushort id)
+        {
+            _id = id;
+
+            // get token type
+            switch (token.TokenType)
+            {
+                case TokenType.AssemblyRef:
+                    _clrTable = NanoClrTable.TBL_AssemblyRef;
+                    break;
+
+                case TokenType.TypeRef:
+                    _clrTable = NanoClrTable.TBL_TypeRef;
+                    break;
+
+                case TokenType.Field:
+                    _clrTable = NanoClrTable.TBL_FieldDef;
+                    break;
+
+                case TokenType.Method:
+                    _clrTable = NanoClrTable.TBL_MethodDef;
+                    break;
+
+                case TokenType.TypeDef:
+                    _clrTable = NanoClrTable.TBL_TypeDef;
+                    break;
+
+                //case TokenType.MethodSpec:
```

Comment:
on purpose? And 2 times the same one. From previous point, most likely te MethodSpec and MethodDef not implemented, so let's be explicit in those commented code if it's about this!